pkg/asset/installconfig/gcp/validation.go

```diff
@@ -16,10 +17,33 @@ func Validate(client API, ic *types.InstallConfig) error {
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("gcp"))...)
+	allErrs = append(allErrs, validatePublicZoneID(client, ic, field.NewPath("platform").Child("gcp"))...)
 
 	return allErrs.ToAggregate()
 }
 
+func validatePublicZoneID(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
```

Comment:
This is duplicate of the validation code in `pkg/types`. We don't need to duplicate here.

Not sure if it is in scope for this PR but any validation here would be using the platform SDK to determine whether the value is valid.

Reply:
Nuked from orbit!